src/webui/utils/api.js

```diff
@@ -1,49 +1,84 @@
+const BASE_URL = window.VERDACCIO_API_URL;
 
-      const token = storage.getItem('token');
-      if (token) {
-        if (!options.headers) options.headers = {};
+/**
+ * Handles response according to content type
+ * @param {object} response
+ * @returns {promise}
+ */
+function handleResponseType(response) {
+  if (response.headers) {
+    const contentType = response.headers.get('Content-Type');
+    if (contentType.includes('application/json')) {
+      return Promise.all([response.ok, response.json()]);
+    }
+    // it includes all text types
+    if (contentType.includes('text/')) {
+      return Promise.all([response.ok, response.text()]);
+    }
+  }
+};
 
-        options.headers.authorization = `Bearer ${token}`;
-      }
+/**
+ * checkAuthentication if 401 then clears storage
+ */
+function checkAuthentication(response) {
+  if (response.status === HTTP_STATUS.UNAUTHORIZED) {
+    storage.removeItem('token');
+    storage.removeItem('username');
+    /**
+     * @todo 
+     * Remove it when global storage available (e.g. Redux) which can re-render
+     * application without reloading it forcefully.
+     */
+    window.location.reload();
+  }
+  return response;
+}
 
-      if (!['http://', 'https://', '//'].some((prefix) => url.startsWith(prefix))) {
-        url = window.VERDACCIO_API_URL + url;
-      }
+/**
+ * API
+ */
+class API {
+  constructor(baseUrl) {
+    this.baseUrl = baseUrl;
+  }
 
-      /**
-       * Handles response according to content type
-       * @param {object} response
-       * @returns {promise}
-       */
-      function handleResponseType(response) {
-        if (response.headers) {
-          const contentType = response.headers.get('Content-Type');
-          if (contentType.includes('application/pdf')) {
-            return Promise.all([response.ok, response.blob()]);
-          }
-          if (contentType.includes('application/json')) {
-            return Promise.all([response.ok, response.json()]);
-          }
-          // it includes all text types
-          if (contentType.includes('text/')) {
-            return Promise.all([response.ok, response.text()]);
-          }
-        }
+  request(resource = '', method = 'GET', options = {}) {
+    let endpoint;
+    
+    // checks for token
+    const token = storage.getItem('token');
+    if (token) {
+      if (!options.headers) {
+        options.headers = {};
       }
+      options.headers.authorization = `Bearer ${token}`;
+    }
 
-      return new Promise((resolve, reject) => {
-        fetch(url, {
-          method,
-          credentials: 'same-origin',
-          ...options
-        })
+    // Checks if resource is already a URL
+    const protocols = ['http://', 'https://', '//'];
+    const isResource = protocols.some((prefix) => resource.startsWith(prefix));
+    if (isResource === false) {
+      endpoint = this.baseUrl + resource;
+    }
+
+    // Final Promise
+    return new Promise(function APIPromise(resolve, reject){
```

Comment:
`eslint` should enforce `){` space between those characters.

Reply:
or prettify, in any case, we have to check that in the future eslint PR. So, just check but don't change it. I want this fails when I apply the proper rules.